#!/bin/bash
#PBS -P pq84
#PBS -q normal 
#PBS -N Variant_calling_consensus
#PBS -j oe
#PBS -m ae
#PBS -l walltime=24:00:00,mem=10GB,ncpus=5
#PBS -l storage=gdata/pq84+scratch/pq84
#PBS -M jacob.westaway@menzies.edu.au

echo "---------------------------------------"
echo "PBS: Job identifier is $PBS_JOBID"
echo "PBS: Job name is $PBS_JOBNAME"

echo "---------------------------------------"
echo "Define paths"
module load bcftools/1.12
module load R/4.1.0 
export R_LIBS_USER="/g/data/pq84/R"

echo "---------------------------------------"
echo "---------------------------------------"
echo 'Change to working directory and set env variables'
PICARD="/g/data/pq84/bin/picard/build/libs/picard.jar"
INDEXTDIR="/g/data/pq84/malaria/data/ref_genomes/PKA1H1/fasta/strain_A1_H.1.Icor.fasta"
OUTDIR="/g/data/pq84/malaria/outputs/04_Variant_calling/consensus/"
INDIR="/g/data/pq84/malaria/outputs/04_Variant_calling/"

cd /g/data/pq84/malaria/

echo "---------------------------------------------------------------------------------------------------------------------"
echo "Create consensus VCF"
echo "---------------------------------------------------------------------------------------------------------------------"

echo "---------------------------------------"
echo 'Merge variants from the two variant callers'
bcftools merge --threads 5 --force-samples -o $OUTDIR/merged_variants_only.vcf.gz $INDIR/gatk/GATK_genotyped.vcf.gz $INDIR/bcftools/bcftools_genotyped.vcf.gz

echo "---------------------------------------"
echo "Query VCF files to get a list of 'variant IDs' from each tool"
bcftools query -f '%CHROM %POS %ID %REF %ALT [\t%SAMPLE DP=%DP GQ=%GQ MQ=%MQ PL=%PL]\n' $INDIR/gatk/GATK_genotyped.vcf.gz > $OUTDIR/GATK_query.tsv
bcftools query -f '%CHROM %POS %ID %REF %ALT [\t%SAMPLE DP=%DP GQ=%GQ MQ=%MQ PL=%PL]\n' $INDIR/bcftools/bcftools_genotyped.vcf.gz > $OUTDIR/bcftools_query.tsv

echo "---------------------------------------"
echo "Execute R script - performs an inner join on the 'variant IDs' above to find the overlapping variants"
Rscript /g/data/pq84/malaria/scripts/pipeline/04_Variant_calling/vcf_wrangle.R

echo "---------------------------------------"
echo "Create a seperate file that contains all the header information for the vcf"
cd $OUTDIR

bcftools view merged_variants_only.vcf.gz | sed '/#CHROM/q' > vcf_head.vcf

echo "---------------------------------------"
echo "Remove the column names from the variant names document to create a list of grep patterns"
tail -n +2 vcf_variant_names.tsv > grep_patterns.txt 

echo "---------------------------------------"
echo "Filter for variants from the original vcf that are called by both callers by using grep to match patterns created above" 
bcftools view merged_variants_only.vcf.gz | fgrep -f grep_patterns.txt - > filtered.vcf

echo "---------------------------------------"
echo "Concatenate the vcf header in the variants called by both callers"
cat vcf_head.vcf filtered.vcf > Consensus.vcf

echo "---------------------------------------"
echo "Finished "
