#!/bin/bash
#PBS -P pq84
#PBS -q normal 
#PBS -N Annotate_variants
#PBS -j oe
#PBS -m ae
#PBS -l walltime=12:00:00,mem=80GB,ncpus=10
#PBS -l storage=gdata/pq84+scratch/pq84
#PBS -M jacob.westaway@menzies.edu.au

echo "---------------------------------------"
echo "PBS: Job identifier is $PBS_JOBID"
echo "PBS: Job name is $PBS_JOBNAME"

echo "---------------------------------------"
echo "Define paths and load modules"
export PATH=$PATH:/g/data/pq84/bin/ensembl-vep

echo "---------------------------------------"
echo "---------------------------------------"
echo 'Change to working directory and set env variables'
INDIR="/g/data/pq84/malaria/outputs/04_Variant_calling/filtered/"
OUTDIR="/g/data/pq84/malaria/outputs/05_Variant_annotation"
GFF="/g/data/pq84/malaria/data/ref_genomes/PKA1H1/gff/updated_version"
FASTA="/g/data/pq84/malaria/data/ref_genomes/PKA1H1/fasta/strain_A1_H.1.Icor.fasta"

cd $INDIR

echo "---------------------------------------------------------------------------------------------------------------------"
echo "SNPs"
vep -i FILTERED_SNPs.vcf --gff $GFF/PlasmoDB-55_PknowlesiA1H1.gff.gz --fasta $FASTA -o $OUTDIR/SNP_variant_annotation.txt --force_overwrite --everything

echo "---------------------------------------------------------------------------------------------------------------------"
echo "INDELs"
vep -i FILTERED_INDELs.vcf --gff $GFF/PlasmoDB-55_PknowlesiA1H1.gff.gz --fasta $FASTA -o $OUTDIR/Indel_variant_annotation.txt --force_overwrite --everything

echo "---------------------------------------"
echo "FINISHED"